### PR TITLE
fix: Switch from rollup to webpack for bundling JS

### LINKED ISSUE #385: Builds have been failing in CircleCI
[Builds have been failing in CircleCI](https://circleci.com/gh/bwinton/SnoozeTabs/641) for awhile now. Seems to have been a repeat problem with rollup for bundling our JS, not sure why. Going to take a crack at just replacing it with Webpack, now that I'm more comfortable with it

### PR DESCRIPTION
**Pull request by lmorchard**
- Remove Rollup, related dependencies, and config

- Add Webpack, related dependencies, and config

- Add webpack.config.js and tweak npm scripts for build process

- Upgrade from babel-preset-es2015 to babel-preset-env

- Upgrade expected Node version to 8.9.4

- Add addon.xpi build artifact to .gitignore

Fixes #385.

### CODE REVIEW

**Comment by lmorchard** [comment]
This should fix the breaking builds on CircleCI, since rollup seems to have often been the culprit.

There's more we could do with Webpack here (i.e. processing .scss), but I wanted to avoid a rabbit hole and limit the changes to JS and replacing rollup

**Comment by lmorchard** [comment]
Oh yeah, and nothing should change in terms of what commands you run to build the project. Might have to get a newer version of node (8.9.4) if you don't already have it and do an npm install

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const NODE_ENV = process.env.NODE_ENV || 'development';
+
+module.exports = {
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: '[name].js'
+  },
+  entry: {
+    background: './src/background.js',
```
Comment (bwinton) [inline webpack.config.js]: Should we throw `''`s around this, for consistency?

Comment (lmorchard) [reply webpack.config.js]: That actually fails linting 😞 It requires quotes for the keys with slashes, but gripes about quotes where they're "not needed"

Comment (bwinton) [reply webpack.config.js]: 😂 ¯\_(ツ)_/¯

**Pull request merged**